src/bluesky/utils/__init__.py

```diff
@@ -1963,8 +1964,43 @@ def plan(plan):
         Wrapped plans
     """
 
-    @wraps(plan)
+    @wraps_plan(bs_plan, plan)
     def wrapper(*args, **kwargs) -> Plan:
-        return Plan(plan, *args, **kwargs)
+        return Plan(bs_plan, *args, **kwargs)
```

Comment:
If we are going to do this wouldn't be simpler to directly set it?

```suggestion
        return Plan(bs_plan, *args, **kwargs)

    wrapper.__is_plan__ = True
```

Reply:
If I do this, it seems that `mv.__wrapped__` is marked with `__is_plan__`, but not `mv` itself:

```
In [11]: is_plan(mv)
Out[11]: False

In [12]: is_plan(mv.__wrapped__)
Out[12]: True
```

Reply:
Edit: Actually, I had some weird state. I tried this again, and it appears to work, and removes the need for the customized `wrap` and `mark_as_plan` functions. If we decide to go with the marker this is probably the way to go.